### PR TITLE
Vendor clicks link on payment receipt email, goes to auction show page

### PR DESCRIPTION
**Pull request by jessieay**
* Part of #978
* This PR adds the route for the receipt, includes receipt link in email
  notification, and adds controller logic that does not allow logged out
  user or non-winner to visit this page.
* Winning vendor cannot visit this page unless the auction is paid
* Page does not have content yet

### CODE REVIEW

**Comment by adelevie** [comment]
Looks good to me. I can merge or let you add more stuff to this?

**Comment by jessieay** [comment]
Adjusted some stuff that was making tests pass -- this is good to go for now, alan, I plan on adding the next part (page content) in a separate PR

File: spec/controllers/receipts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+          expect(response).to be_redirect
+          expect(response.location).to eq('http://test.host/')
+        end
+      end
+    end
+
+    context 'current user is not winning vendor' do
+      it 'redirects the user' do
+          auction = create(:auction, :paid)
```
Comment (adelevie) [inline spec/controllers/receipts_controller_spec.rb]: too much spacing?

File: spec/controllers/receipts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          get :new, { auction_id: auction.id }, user_id: current_user.id
+
+          expect(response).to be_redirect
+          expect(response.location).to eq('http://test.host/')
+      end
+    end
+
+    context 'current user is a guest' do
+      it 'redirects the user' do
+          auction = create(:auction, :paid)
```
Comment (adelevie) [inline spec/controllers/receipts_controller_spec.rb]: spacing

File: spec/controllers/receipts_controller_spec.rb (hunk 1 above)
Comment (jessieay) [inline spec/controllers/receipts_controller_spec.rb]: you are right!

**Pull request merged**